test/cell_test.rb

```diff
@@ -0,0 +1,80 @@
+require "minitest/autorun"
+require "minitest/pride"
+require "mocha/minitest"
+require "./lib/ship"
+require "./lib/cell"
+
+class CellTest < Minitest::Test
+
+  def test_it_exists
+    cell = Cell.new("B4")
+    assert_instance_of Cell, cell
+  end
+
+  def test_it_has_attributes
+    cell = Cell.new("B4")
+    assert_equal "B4", cell.coordinate
+    assert_nil cell.ship
+    assert_equal true, cell.empty?
+  end
+
+  def test_it_can_place_a_ship
+    cell = Cell.new("B4")
+
+    cruiser = Ship.new("Cruiser", 3)
+
+    cell.place_ship(cruiser)
+    assert_equal cruiser, cell.ship
+
+    assert_equal false, cell.empty?
+  end
+
+  def test_it_can_be_fired_upon
+    # skip
+    cell = Cell.new("B4")
+
+    cruiser = Ship.new("Cruiser", 3)
+    cell.place_ship(cruiser)
+
+
+    assert_equal false, cell.fired_upon?
+
+    cell.fire_upon
+
+    assert_equal 2, cell.ship.health
+
+    assert_equal true, cell.fired_upon?
+
+  end
+
+  def test_it_can_render
+    cell_1 = Cell.new("B4")
+    assert_equal ".", cell_1.render
+    cell_1.fire_upon
+    assert_equal "M", cell_1.render
+
+    cell_2 = Cell.new("C3")
+    cruiser = Ship.new("Cruiser", 3)
+
+    cell_2.place_ship(cruiser)
+
+    assert_equal ".", cell_2.render
+    assert_equal "S", cell_2.render(true)
+
+    cell_2.fire_upon
+
+    assert_equal "H", cell_2.render
+    assert_equal false, cruiser.sunk?
+
+    cruiser.hit
+    cruiser.hit
+
+    assert_equal "X", cell_2.render
+    assert_equal true, cruiser.sunk?
+
+
+  end
+
+
+
+end
```

Comment:
First Pry Session (Testing):
[1] pry(main)> require './lib/ship'
=> true
[2] pry(main)> require './lib/cell'
=> true
[3] pry(main)> cell = Cell.new("B4")
=> #<Cell:0x00007fd84282f790
 @coordinate="B4",
 @empty=true,
 @fired_upon=false,
 @ship=nil>
[4] pry(main)> cruiser = Ship.new("Cruiser", 3)
=> #<Ship:0x00007fd840b5ae80 @health=3, @length=3, @name="Cruiser">
[5] pry(main)> cell.place_ship(cruiser)
=> #<Ship:0x00007fd840b5ae80 @health=3, @length=3, @name="Cruiser">
[6] pry(main)> cell.fired_upon?
=> false
[7] pry(main)> cell.fire_upon
=> 2
[8] pry(main)> cell.ship_health
NoMethodError: undefined method `ship_health' for #<Cell:0x00007fd84282f790>
from (pry):8:in `__pry__'
[9] pry(main)> cell.ship.health
=> 2
[10] pry(main)> cell.fired_upon?
=> true

Reply:
Second Pry Session (Testing):
[1] pry(main)> require './lib/ship'
=> true
[2] pry(main)> require './lib/cell'
=> true
[3] pry(main)> cell_1 = Cell.new("B4")
=> #<Cell:0x00007ff7f117f188
 @coordinate="B4",
 @empty=true,
 @fired_upon=false,
 @ship=nil>
[4] pry(main)> cell_1.render
=> "."
[5] pry(main)> cell_1.fire_upon
=> nil
[6] pry(main)> cell_1.render
=> "M"
[7] pry(main)> cell_2 = Cell.new("C3")
=> #<Cell:0x00007ff7f1176600
 @coordinate="C3",
 @empty=true,
 @fired_upon=false,
 @ship=nil>
[8] pry(main)> cruiser = Ship.new("Cruiser", 3)
=> #<Ship:0x00007ff7f2903ac0 @health=3, @length=3, @name="Cruiser">
[9] pry(main)> cell_2.place_ship(cruiser)
=> #<Ship:0x00007ff7f2903ac0 @health=3, @length=3, @name="Cruiser">
[10] pry(main)> cell_2.render
=> "."
[11] pry(main)> cell_2.render(true)
=> "S"
[12] pry(main)> cell_2.fire_upon
=> 2
[13] pry(main)> cell_2.render
=> "H"
[14] pry(main)> cruiser.sunk?
=> false
[15] pry(main)> cruiser.hit
=> 1
[16] pry(main)> cruiser.hit
=> 0
[17] pry(main)> cruiser.sunk?
=> true
[18] pry(main)> cell_2.render
=> "X"

Reply:
Some of our pry info is slightly different than interaction pattern (the interaction pattern returns nothing whereas ours returns something)

INSTANCES:
1. place_ship(cruiser).  -> I can go check ship class
2. fire_upon
3. .hit

I don't know if this due to pry session, do we need to update our code, does it matter, etc...?

Reply:
Good morning Priya,  I just reviewed all of your comments, but it looks like some of those points were resolved.  So lets focus on this one

I've never checked something like that where our pry sessions shouldn't have return values for these specific methods.  I have typically just been focused on functionality.  

I'm mostly concerned that each method is in fact placing the object or behaving in a way that is predictable.  

I am always willing to challenge my assumptions, but typically if the behavior brought our tests to green, I have moved on.  

I usually use the pry interactions to guide my tests (we don't need to put an assert_equal for `cruiser.hit`, but we do need to put an assert_equal on the method `cell_2.render` returns "X"

Does that make sense?

Reply:
[Here's a doc that I found](http://ruby-for-beginners.rubymonstas.org/writing_methods/return_values.html) when I searched for `how to have methods not return values in pry iteraction ruby`. I think we'll be good.